scapy/tools/UTscapy/__init__.py

```diff
@@ -750,9 +731,7 @@ def main(argv):
                     raise getopt.GetoptError("Unknown output format %s" % msg)
                 TESTFILES = resolve_testfiles(TESTFILES)
             elif opt == "-o":
-                OUTPUTFILE = open(optarg, "wb")
-            elif opt == "-l":
-                LOCAL = 1
+                OUTPUTFILE = optarg
```

Comment:
I understand that you want to open the OUTPUTFILE at the very last moment, to avoid keeping the pointers for long.

Problem is that if the OUTPUTFILE is not writable, UTScapy will first execute all the tests then fail sadly while writing them, which can be frustrating.

Reply:
I don't understand your issue. It works perfectly for me when I execute `run_tests` :-/
 OUTPUTFILE is opened in write mode if it is not `stdout` (to avoid closing it) when a `write` operation is required which is at the very end of the `main` function